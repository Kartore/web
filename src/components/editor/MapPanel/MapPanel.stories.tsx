import { MapPanel } from '.';
import { Meta, StoryObj } from '@storybook/react';
import { osmLiberty } from '~/samples/osm-liberty.ts';

const meta = {
  component: MapPanel,
} satisfies Meta<typeof MapPanel>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    style: {
      height: '500px',
    },
    mapStyle: osmLiberty,
  },
};
