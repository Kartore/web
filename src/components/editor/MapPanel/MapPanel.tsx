import 'maplibre-gl/dist/maplibre-gl.css';
import type { CSSProperties, FC } from 'react';

import type { StyleSpecification } from 'maplibre-gl';
import { Map } from 'react-map-gl/maplibre';

export const MapPanel: FC<{
  style?: CSSProperties;
  mapStyle: StyleSpecification;
}> = ({ style, mapStyle }) => {
  return (
    <Map
      initialViewState={{
        longitude: 139.767,
        latitude: 35.681,
        zoom: 15,
      }}
      style={{ height: '100%', ...style }}
      mapStyle={mapStyle}
      maplibreLogo
    />
  );
};
