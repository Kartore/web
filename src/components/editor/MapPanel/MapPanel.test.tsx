import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './MapPanel.stories';

const { Default } = composeStories(Stories);

describe('Component: MapPanel', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
