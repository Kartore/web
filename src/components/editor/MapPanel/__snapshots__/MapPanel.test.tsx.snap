// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: MapPanel > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    style="position: relative; width: 100%; height: 500px;"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
